sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/aio/_azure_appconfiguration_client_async.py

```diff
@@ -434,6 +434,60 @@ async def delete_configuration_setting(
                 key="MyKey", label="MyLabel"
             )
         """
+
+    @overload
+    async def delete_configuration_setting(  # pylint:disable=delete-operation-wrong-return-type
+        self,
+        configuration_setting: ConfigurationSetting,
+        *,
+        etag: Optional[str] = None,
+        match_condition: MatchConditions = MatchConditions.Unconditionally,
+        **kwargs: Any,
+    ) -> ConfigurationSetting:
+        """Delete a ConfigurationSetting if it exists
+
+        :param configuration_setting: the ConfigurationSetting to be deleted (if not exists)
+        :type configuration_setting: ~azure.appconfiguration.ConfigurationSetting
+        :keyword str etag: check if the ConfigurationSetting is changed. Set None to skip checking etag
+        :keyword match_condition: The match condition to use upon the etag.
+            The default value is MatchConditions.Unconditionally.
+        :paramtype match_condition: ~azure.core.MatchConditions
+        :return: The deleted ConfigurationSetting returned from the service, or None if it doesn't exist.
+        :rtype: ~azure.appconfiguration.ConfigurationSetting
+        :raises: :class:`~azure.core.exceptions.HttpResponseError`, \
+            :class:`~azure.core.exceptions.ClientAuthenticationError`, \
+            :class:`~azure.core.exceptions.ResourceReadOnlyError`, \
+            :class:`~azure.core.exceptions.ResourceModifiedError`, \
+            :class:`~azure.core.exceptions.ResourceNotModifiedError`, \
+            :class:`~azure.core.exceptions.ResourceNotFoundError`, \
+            :class:`~azure.core.exceptions.ResourceExistsError`
+
+        Example
+
+        .. code-block:: python
+
+            config_setting = ConfigurationSetting(key="MyKey", label="MyLabel")
+            deleted_config_setting = await client.delete_configuration_setting(configuration_setting=config_setting)
+        """
+
+    @distributed_trace_async
+    async def delete_configuration_setting(self, *args, **kwargs) -> ConfigurationSetting:
+        try:
+            config_setting = kwargs.pop("configuration_setting", None)
+            if not config_setting:
```

Comment:
Do we need to check the length of the args?

Reply:
Not necessary. The `args[0]` of the two overloads are required and in different types. We assume the `args[0]` is in type `ConfigurationSetting` at beginning, it will throw an `AttributeError` once its type is in `str`.

Reply:
But you have

def delete_configuration_setting(  # pylint:disable=delete-operation-wrong-return-type
        self,
        key: str,
        label: Optional[str] = None,
        *,
        etag: Optional[str] = None,
        match_condition: MatchConditions = MatchConditions.Unconditionally,
        **kwargs,
        **kwargs: Any,
    ) -> ConfigurationSetting:

too?

Why args[0] is ConfigurationSetting?

Reply:
args[0] could be in type `ConfigurationSetting` or `str`. If the args[0] is not in type `ConfigurationSetting`, will try to get values as the overload you pasted in the `except` block.

Reply:
What is the expected behavior if user uses delete_configuration_setting(<a valid config_setting>, <a label>)?

Reply:
Now will throw error: `IndexError: tuple index out of range`. 
I think the expected behavior would be throwing a similar error like without the overload, would be
```
TypeError: AzureAppConfigurationClient.delete_configuration_setting() missing 1 required positional argument: 'key' or 'configuration_setting'
```
Do you think the error info is clear? Any thoughts?